lambdaclass/mirra_backend
# [GH-429] Dungeon Settlement Levels

## Motivation

Adds DungeonSettlementLevels to Users
Closes https://github.com/lambdaclass/champions_of_mirra/issues/429

## Summary of changes

- Creates DungeonSettlementLevel schema
- Removes kaline_tree_id requirement in user creation
- Creates message for claiming Dungeon Settlement AFK Rewards
  - Did not create a message for _showing_ these rewards because we will be automatically claiming them everytime we open the Dungeon Settlement screen.
- Rename most Kalline Tree AFK Rewards messages and funcs because we now have two different AFK Reward groups
- Add Tests

## How to test it?

You can create a user and check that his dungeon_settlement_level is instantiated at level 1.

If you give your user 50 "Blueprints" (using `GameBackend.Users.Currencies.add_currency_by_name!/3`), you will be able to level up the dungeon settlement of your user with `Champions.Users.level_up_dungeon_settlement/1`. If you get your user again (`Champions.Users.get_user_by_username/1`) you will be able to see that his dungeon_settlement_level has increased, and his "Gold" and "Blueprints" currencies decreased!

Since this implements protobuf messages, this PR also adds SocketTester messages! You can also try those :P
